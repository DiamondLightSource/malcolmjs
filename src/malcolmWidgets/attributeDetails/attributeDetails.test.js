import React from 'react';
import { createMount } from 'material-ui/test-utils';
import AttributeDetails from './attributeDetails.component';

describe('AttributeDetails', () => {
  let mount;

  beforeEach(() => {
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('renders correctly', () => {
    const attribute = {
      name: 'Attribute 1',
      alarm: {
        severity: 0,
      },
    };

    const wrapper = mount(
      <AttributeDetails attribute={attribute}>
        <div>value</div>
      </AttributeDetails>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('shows that it is unable to render unprocessed attributes', () => {
    const attribute = {
      name: 'Attribute 1',
      alarm: {
        severity: 0,
      },
      unableToProcess: true,
    };

    const wrapper = mount(
      <AttributeDetails attribute={attribute}>
        <div>value</div>
      </AttributeDetails>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
