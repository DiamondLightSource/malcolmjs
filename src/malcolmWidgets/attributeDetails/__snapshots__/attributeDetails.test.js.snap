// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributeDetails renders correctly 1`] = `
<WithStyles(AttributeDetails)
  attribute={
    Object {
      "alarm": Object {
        "severity": 0,
      },
      "name": "Attribute 1",
    }
  }
>
  <AttributeDetails
    attribute={
      Object {
        "alarm": Object {
          "severity": 0,
        },
        "name": "Attribute 1",
      }
    }
    classes={
      Object {
        "controlContainer": "AttributeDetails-controlContainer-3",
        "div": "AttributeDetails-div-1",
        "textName": "AttributeDetails-textName-2",
      }
    }
    theme={
      Object {
        "breakpoints": Object {
          "between": [Function],
          "down": [Function],
          "keys": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "only": [Function],
          "up": [Function],
          "values": Object {
            "lg": 1280,
            "md": 960,
            "sm": 600,
            "xl": 1920,
            "xs": 0,
          },
          "width": [Function],
        },
        "direction": "ltr",
        "mixins": Object {
          "gutters": [Function],
          "toolbar": Object {
            "@media (min-width:0px) and (orientation: landscape)": Object {
              "minHeight": 48,
            },
            "@media (min-width:600px)": Object {
              "minHeight": 64,
            },
            "minHeight": 56,
          },
        },
        "overrides": Object {},
        "palette": Object {
          "action": Object {
            "active": "rgba(0, 0, 0, 0.54)",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "disabledBackground": "rgba(0, 0, 0, 0.12)",
            "hover": "rgba(0, 0, 0, 0.08)",
            "hoverOpacity": 0.08,
            "selected": "rgba(0, 0, 0, 0.14)",
          },
          "augmentColor": [Function],
          "background": Object {
            "default": "#fafafa",
            "paper": "#fff",
          },
          "common": Object {
            "black": "#000",
            "white": "#fff",
          },
          "contrastThreshold": 3,
          "divider": "rgba(0, 0, 0, 0.12)",
          "error": Object {
            "contrastText": "#fff",
            "dark": "#d32f2f",
            "light": "#e57373",
            "main": "#f44336",
          },
          "getContrastText": [Function],
          "grey": Object {
            "100": "#f5f5f5",
            "200": "#eeeeee",
            "300": "#e0e0e0",
            "400": "#bdbdbd",
            "50": "#fafafa",
            "500": "#9e9e9e",
            "600": "#757575",
            "700": "#616161",
            "800": "#424242",
            "900": "#212121",
            "A100": "#d5d5d5",
            "A200": "#aaaaaa",
            "A400": "#303030",
            "A700": "#616161",
          },
          "primary": Object {
            "contrastText": "#fff",
            "dark": "#303f9f",
            "light": "#7986cb",
            "main": "#3f51b5",
          },
          "secondary": Object {
            "contrastText": "#fff",
            "dark": "#c51162",
            "light": "#ff4081",
            "main": "#f50057",
          },
          "text": Object {
            "disabled": "rgba(0, 0, 0, 0.38)",
            "hint": "rgba(0, 0, 0, 0.38)",
            "primary": "rgba(0, 0, 0, 0.87)",
            "secondary": "rgba(0, 0, 0, 0.54)",
          },
          "tonalOffset": 0.2,
          "type": "light",
        },
        "props": Object {},
        "shadows": Array [
          "none",
          "0px 1px 3px 0px rgba(0, 0, 0, 0.2),0px 1px 1px 0px rgba(0, 0, 0, 0.14),0px 2px 1px -1px rgba(0, 0, 0, 0.12)",
          "0px 1px 5px 0px rgba(0, 0, 0, 0.2),0px 2px 2px 0px rgba(0, 0, 0, 0.14),0px 3px 1px -2px rgba(0, 0, 0, 0.12)",
          "0px 1px 8px 0px rgba(0, 0, 0, 0.2),0px 3px 4px 0px rgba(0, 0, 0, 0.14),0px 3px 3px -2px rgba(0, 0, 0, 0.12)",
          "0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px 0px rgba(0, 0, 0, 0.14),0px 1px 10px 0px rgba(0, 0, 0, 0.12)",
          "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 5px 8px 0px rgba(0, 0, 0, 0.14),0px 1px 14px 0px rgba(0, 0, 0, 0.12)",
          "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px 0px rgba(0, 0, 0, 0.14),0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
          "0px 4px 5px -2px rgba(0, 0, 0, 0.2),0px 7px 10px 1px rgba(0, 0, 0, 0.14),0px 2px 16px 1px rgba(0, 0, 0, 0.12)",
          "0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12)",
          "0px 5px 6px -3px rgba(0, 0, 0, 0.2),0px 9px 12px 1px rgba(0, 0, 0, 0.14),0px 3px 16px 2px rgba(0, 0, 0, 0.12)",
          "0px 6px 6px -3px rgba(0, 0, 0, 0.2),0px 10px 14px 1px rgba(0, 0, 0, 0.14),0px 4px 18px 3px rgba(0, 0, 0, 0.12)",
          "0px 6px 7px -4px rgba(0, 0, 0, 0.2),0px 11px 15px 1px rgba(0, 0, 0, 0.14),0px 4px 20px 3px rgba(0, 0, 0, 0.12)",
          "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 12px 17px 2px rgba(0, 0, 0, 0.14),0px 5px 22px 4px rgba(0, 0, 0, 0.12)",
          "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 13px 19px 2px rgba(0, 0, 0, 0.14),0px 5px 24px 4px rgba(0, 0, 0, 0.12)",
          "0px 7px 9px -4px rgba(0, 0, 0, 0.2),0px 14px 21px 2px rgba(0, 0, 0, 0.14),0px 5px 26px 4px rgba(0, 0, 0, 0.12)",
          "0px 8px 9px -5px rgba(0, 0, 0, 0.2),0px 15px 22px 2px rgba(0, 0, 0, 0.14),0px 6px 28px 5px rgba(0, 0, 0, 0.12)",
          "0px 8px 10px -5px rgba(0, 0, 0, 0.2),0px 16px 24px 2px rgba(0, 0, 0, 0.14),0px 6px 30px 5px rgba(0, 0, 0, 0.12)",
          "0px 8px 11px -5px rgba(0, 0, 0, 0.2),0px 17px 26px 2px rgba(0, 0, 0, 0.14),0px 6px 32px 5px rgba(0, 0, 0, 0.12)",
          "0px 9px 11px -5px rgba(0, 0, 0, 0.2),0px 18px 28px 2px rgba(0, 0, 0, 0.14),0px 7px 34px 6px rgba(0, 0, 0, 0.12)",
          "0px 9px 12px -6px rgba(0, 0, 0, 0.2),0px 19px 29px 2px rgba(0, 0, 0, 0.14),0px 7px 36px 6px rgba(0, 0, 0, 0.12)",
          "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 20px 31px 3px rgba(0, 0, 0, 0.14),0px 8px 38px 7px rgba(0, 0, 0, 0.12)",
          "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 21px 33px 3px rgba(0, 0, 0, 0.14),0px 8px 40px 7px rgba(0, 0, 0, 0.12)",
          "0px 10px 14px -6px rgba(0, 0, 0, 0.2),0px 22px 35px 3px rgba(0, 0, 0, 0.14),0px 8px 42px 7px rgba(0, 0, 0, 0.12)",
          "0px 11px 14px -7px rgba(0, 0, 0, 0.2),0px 23px 36px 3px rgba(0, 0, 0, 0.14),0px 9px 44px 8px rgba(0, 0, 0, 0.12)",
          "0px 11px 15px -7px rgba(0, 0, 0, 0.2),0px 24px 38px 3px rgba(0, 0, 0, 0.14),0px 9px 46px 8px rgba(0, 0, 0, 0.12)",
        ],
        "spacing": Object {
          "unit": 8,
        },
        "transitions": Object {
          "create": [Function],
          "duration": Object {
            "complex": 375,
            "enteringScreen": 225,
            "leavingScreen": 195,
            "short": 250,
            "shorter": 200,
            "shortest": 150,
            "standard": 300,
          },
          "easing": Object {
            "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
            "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
            "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
            "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
          },
          "getAutoHeightDuration": [Function],
        },
        "typography": Object {
          "body1": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 400,
            "lineHeight": "1.46429em",
          },
          "body2": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "lineHeight": "1.71429em",
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "textTransform": "uppercase",
          },
          "caption": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.75rem",
            "fontWeight": 400,
            "lineHeight": "1.375em",
          },
          "display1": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "2.125rem",
            "fontWeight": 400,
            "lineHeight": "1.20588em",
          },
          "display2": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "2.8125rem",
            "fontWeight": 400,
            "lineHeight": "1.06667em",
            "marginLeft": "-.02em",
          },
          "display3": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "3.5rem",
            "fontWeight": 400,
            "letterSpacing": "-.02em",
            "lineHeight": "1.30357em",
            "marginLeft": "-.02em",
          },
          "display4": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "7rem",
            "fontWeight": 300,
            "letterSpacing": "-.04em",
            "lineHeight": "1.14286em",
            "marginLeft": "-.04em",
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": 14,
          "fontWeightLight": 300,
          "fontWeightMedium": 500,
          "fontWeightRegular": 400,
          "headline": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1.5rem",
            "fontWeight": 400,
            "lineHeight": "1.35417em",
          },
          "pxToRem": [Function],
          "round": [Function],
          "subheading": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1rem",
            "fontWeight": 400,
            "lineHeight": "1.5em",
          },
          "title": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1.3125rem",
            "fontWeight": 500,
            "lineHeight": "1.16667em",
          },
        },
        "zIndex": Object {
          "appBar": 1100,
          "drawer": 1200,
          "mobileStepper": 1000,
          "modal": 1300,
          "snackbar": 1400,
          "tooltip": 1500,
        },
      }
    }
  >
    <div>
      <div
        className="AttributeDetails-div-1"
      >
        <AttributeAlarm
          alarmSeverity={0}
        >
          <pure(InfoOutline)
            nativeColor="#72a7cf"
          >
            <InfoOutline
              nativeColor="#72a7cf"
            >
              <WithStyles(SvgIcon)
                nativeColor="#72a7cf"
              >
                <SvgIcon
                  classes={
                    Object {
                      "colorAction": "MuiSvgIcon-colorAction-7",
                      "colorDisabled": "MuiSvgIcon-colorDisabled-8",
                      "colorError": "MuiSvgIcon-colorError-9",
                      "colorPrimary": "MuiSvgIcon-colorPrimary-5",
                      "colorSecondary": "MuiSvgIcon-colorSecondary-6",
                      "root": "MuiSvgIcon-root-4",
                    }
                  }
                  color="inherit"
                  nativeColor="#72a7cf"
                  viewBox="0 0 24 24"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-4"
                    color="#72a7cf"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g>
                      <path
                        d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
                      />
                    </g>
                  </svg>
                </SvgIcon>
              </WithStyles(SvgIcon)>
            </InfoOutline>
          </pure(InfoOutline)>
        </AttributeAlarm>
        <WithStyles(Typography)
          className="AttributeDetails-textName-2"
        >
          <Typography
            align="inherit"
            className="AttributeDetails-textName-2"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter-23",
                "alignJustify": "MuiTypography-alignJustify-25",
                "alignLeft": "MuiTypography-alignLeft-22",
                "alignRight": "MuiTypography-alignRight-24",
                "body1": "MuiTypography-body1-19",
                "body2": "MuiTypography-body2-18",
                "button": "MuiTypography-button-21",
                "caption": "MuiTypography-caption-20",
                "colorError": "MuiTypography-colorError-33",
                "colorInherit": "MuiTypography-colorInherit-29",
                "colorPrimary": "MuiTypography-colorPrimary-30",
                "colorSecondary": "MuiTypography-colorSecondary-31",
                "colorTextSecondary": "MuiTypography-colorTextSecondary-32",
                "display1": "MuiTypography-display1-14",
                "display2": "MuiTypography-display2-13",
                "display3": "MuiTypography-display3-12",
                "display4": "MuiTypography-display4-11",
                "gutterBottom": "MuiTypography-gutterBottom-27",
                "headline": "MuiTypography-headline-15",
                "noWrap": "MuiTypography-noWrap-26",
                "paragraph": "MuiTypography-paragraph-28",
                "root": "MuiTypography-root-10",
                "subheading": "MuiTypography-subheading-17",
                "title": "MuiTypography-title-16",
              }
            }
            color="default"
            gutterBottom={false}
            headlineMapping={
              Object {
                "body1": "p",
                "body2": "aside",
                "display1": "h1",
                "display2": "h1",
                "display3": "h1",
                "display4": "h1",
                "headline": "h1",
                "subheading": "h3",
                "title": "h2",
              }
            }
            noWrap={false}
            paragraph={false}
            variant="body1"
          >
            <p
              className="MuiTypography-root-10 MuiTypography-body1-19 AttributeDetails-textName-2"
            >
              Attribute 1
              :
               
            </p>
          </Typography>
        </WithStyles(Typography)>
        <div
          className="AttributeDetails-controlContainer-3"
        >
          <div>
            value
          </div>
        </div>
      </div>
    </div>
  </AttributeDetails>
</WithStyles(AttributeDetails)>
`;

exports[`AttributeDetails shows that it is unable to render unprocessed attributes 1`] = `
<WithStyles(AttributeDetails)
  attribute={
    Object {
      "alarm": Object {
        "severity": 0,
      },
      "name": "Attribute 1",
      "unableToProcess": true,
    }
  }
>
  <AttributeDetails
    attribute={
      Object {
        "alarm": Object {
          "severity": 0,
        },
        "name": "Attribute 1",
        "unableToProcess": true,
      }
    }
    classes={
      Object {
        "controlContainer": "AttributeDetails-controlContainer-36",
        "div": "AttributeDetails-div-34",
        "textName": "AttributeDetails-textName-35",
      }
    }
    theme={
      Object {
        "breakpoints": Object {
          "between": [Function],
          "down": [Function],
          "keys": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "only": [Function],
          "up": [Function],
          "values": Object {
            "lg": 1280,
            "md": 960,
            "sm": 600,
            "xl": 1920,
            "xs": 0,
          },
          "width": [Function],
        },
        "direction": "ltr",
        "mixins": Object {
          "gutters": [Function],
          "toolbar": Object {
            "@media (min-width:0px) and (orientation: landscape)": Object {
              "minHeight": 48,
            },
            "@media (min-width:600px)": Object {
              "minHeight": 64,
            },
            "minHeight": 56,
          },
        },
        "overrides": Object {},
        "palette": Object {
          "action": Object {
            "active": "rgba(0, 0, 0, 0.54)",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "disabledBackground": "rgba(0, 0, 0, 0.12)",
            "hover": "rgba(0, 0, 0, 0.08)",
            "hoverOpacity": 0.08,
            "selected": "rgba(0, 0, 0, 0.14)",
          },
          "augmentColor": [Function],
          "background": Object {
            "default": "#fafafa",
            "paper": "#fff",
          },
          "common": Object {
            "black": "#000",
            "white": "#fff",
          },
          "contrastThreshold": 3,
          "divider": "rgba(0, 0, 0, 0.12)",
          "error": Object {
            "contrastText": "#fff",
            "dark": "#d32f2f",
            "light": "#e57373",
            "main": "#f44336",
          },
          "getContrastText": [Function],
          "grey": Object {
            "100": "#f5f5f5",
            "200": "#eeeeee",
            "300": "#e0e0e0",
            "400": "#bdbdbd",
            "50": "#fafafa",
            "500": "#9e9e9e",
            "600": "#757575",
            "700": "#616161",
            "800": "#424242",
            "900": "#212121",
            "A100": "#d5d5d5",
            "A200": "#aaaaaa",
            "A400": "#303030",
            "A700": "#616161",
          },
          "primary": Object {
            "contrastText": "#fff",
            "dark": "#303f9f",
            "light": "#7986cb",
            "main": "#3f51b5",
          },
          "secondary": Object {
            "contrastText": "#fff",
            "dark": "#c51162",
            "light": "#ff4081",
            "main": "#f50057",
          },
          "text": Object {
            "disabled": "rgba(0, 0, 0, 0.38)",
            "hint": "rgba(0, 0, 0, 0.38)",
            "primary": "rgba(0, 0, 0, 0.87)",
            "secondary": "rgba(0, 0, 0, 0.54)",
          },
          "tonalOffset": 0.2,
          "type": "light",
        },
        "props": Object {},
        "shadows": Array [
          "none",
          "0px 1px 3px 0px rgba(0, 0, 0, 0.2),0px 1px 1px 0px rgba(0, 0, 0, 0.14),0px 2px 1px -1px rgba(0, 0, 0, 0.12)",
          "0px 1px 5px 0px rgba(0, 0, 0, 0.2),0px 2px 2px 0px rgba(0, 0, 0, 0.14),0px 3px 1px -2px rgba(0, 0, 0, 0.12)",
          "0px 1px 8px 0px rgba(0, 0, 0, 0.2),0px 3px 4px 0px rgba(0, 0, 0, 0.14),0px 3px 3px -2px rgba(0, 0, 0, 0.12)",
          "0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px 0px rgba(0, 0, 0, 0.14),0px 1px 10px 0px rgba(0, 0, 0, 0.12)",
          "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 5px 8px 0px rgba(0, 0, 0, 0.14),0px 1px 14px 0px rgba(0, 0, 0, 0.12)",
          "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px 0px rgba(0, 0, 0, 0.14),0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
          "0px 4px 5px -2px rgba(0, 0, 0, 0.2),0px 7px 10px 1px rgba(0, 0, 0, 0.14),0px 2px 16px 1px rgba(0, 0, 0, 0.12)",
          "0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12)",
          "0px 5px 6px -3px rgba(0, 0, 0, 0.2),0px 9px 12px 1px rgba(0, 0, 0, 0.14),0px 3px 16px 2px rgba(0, 0, 0, 0.12)",
          "0px 6px 6px -3px rgba(0, 0, 0, 0.2),0px 10px 14px 1px rgba(0, 0, 0, 0.14),0px 4px 18px 3px rgba(0, 0, 0, 0.12)",
          "0px 6px 7px -4px rgba(0, 0, 0, 0.2),0px 11px 15px 1px rgba(0, 0, 0, 0.14),0px 4px 20px 3px rgba(0, 0, 0, 0.12)",
          "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 12px 17px 2px rgba(0, 0, 0, 0.14),0px 5px 22px 4px rgba(0, 0, 0, 0.12)",
          "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 13px 19px 2px rgba(0, 0, 0, 0.14),0px 5px 24px 4px rgba(0, 0, 0, 0.12)",
          "0px 7px 9px -4px rgba(0, 0, 0, 0.2),0px 14px 21px 2px rgba(0, 0, 0, 0.14),0px 5px 26px 4px rgba(0, 0, 0, 0.12)",
          "0px 8px 9px -5px rgba(0, 0, 0, 0.2),0px 15px 22px 2px rgba(0, 0, 0, 0.14),0px 6px 28px 5px rgba(0, 0, 0, 0.12)",
          "0px 8px 10px -5px rgba(0, 0, 0, 0.2),0px 16px 24px 2px rgba(0, 0, 0, 0.14),0px 6px 30px 5px rgba(0, 0, 0, 0.12)",
          "0px 8px 11px -5px rgba(0, 0, 0, 0.2),0px 17px 26px 2px rgba(0, 0, 0, 0.14),0px 6px 32px 5px rgba(0, 0, 0, 0.12)",
          "0px 9px 11px -5px rgba(0, 0, 0, 0.2),0px 18px 28px 2px rgba(0, 0, 0, 0.14),0px 7px 34px 6px rgba(0, 0, 0, 0.12)",
          "0px 9px 12px -6px rgba(0, 0, 0, 0.2),0px 19px 29px 2px rgba(0, 0, 0, 0.14),0px 7px 36px 6px rgba(0, 0, 0, 0.12)",
          "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 20px 31px 3px rgba(0, 0, 0, 0.14),0px 8px 38px 7px rgba(0, 0, 0, 0.12)",
          "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 21px 33px 3px rgba(0, 0, 0, 0.14),0px 8px 40px 7px rgba(0, 0, 0, 0.12)",
          "0px 10px 14px -6px rgba(0, 0, 0, 0.2),0px 22px 35px 3px rgba(0, 0, 0, 0.14),0px 8px 42px 7px rgba(0, 0, 0, 0.12)",
          "0px 11px 14px -7px rgba(0, 0, 0, 0.2),0px 23px 36px 3px rgba(0, 0, 0, 0.14),0px 9px 44px 8px rgba(0, 0, 0, 0.12)",
          "0px 11px 15px -7px rgba(0, 0, 0, 0.2),0px 24px 38px 3px rgba(0, 0, 0, 0.14),0px 9px 46px 8px rgba(0, 0, 0, 0.12)",
        ],
        "spacing": Object {
          "unit": 8,
        },
        "transitions": Object {
          "create": [Function],
          "duration": Object {
            "complex": 375,
            "enteringScreen": 225,
            "leavingScreen": 195,
            "short": 250,
            "shorter": 200,
            "shortest": 150,
            "standard": 300,
          },
          "easing": Object {
            "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
            "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
            "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
            "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
          },
          "getAutoHeightDuration": [Function],
        },
        "typography": Object {
          "body1": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 400,
            "lineHeight": "1.46429em",
          },
          "body2": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "lineHeight": "1.71429em",
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "textTransform": "uppercase",
          },
          "caption": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.75rem",
            "fontWeight": 400,
            "lineHeight": "1.375em",
          },
          "display1": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "2.125rem",
            "fontWeight": 400,
            "lineHeight": "1.20588em",
          },
          "display2": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "2.8125rem",
            "fontWeight": 400,
            "lineHeight": "1.06667em",
            "marginLeft": "-.02em",
          },
          "display3": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "3.5rem",
            "fontWeight": 400,
            "letterSpacing": "-.02em",
            "lineHeight": "1.30357em",
            "marginLeft": "-.02em",
          },
          "display4": Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "7rem",
            "fontWeight": 300,
            "letterSpacing": "-.04em",
            "lineHeight": "1.14286em",
            "marginLeft": "-.04em",
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": 14,
          "fontWeightLight": 300,
          "fontWeightMedium": 500,
          "fontWeightRegular": 400,
          "headline": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1.5rem",
            "fontWeight": 400,
            "lineHeight": "1.35417em",
          },
          "pxToRem": [Function],
          "round": [Function],
          "subheading": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1rem",
            "fontWeight": 400,
            "lineHeight": "1.5em",
          },
          "title": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1.3125rem",
            "fontWeight": 500,
            "lineHeight": "1.16667em",
          },
        },
        "zIndex": Object {
          "appBar": 1100,
          "drawer": 1200,
          "mobileStepper": 1000,
          "modal": 1300,
          "snackbar": 1400,
          "tooltip": 1500,
        },
      }
    }
  >
    <div>
      <WithStyles(Typography)>
        <Typography
          align="inherit"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter-50",
              "alignJustify": "MuiTypography-alignJustify-52",
              "alignLeft": "MuiTypography-alignLeft-49",
              "alignRight": "MuiTypography-alignRight-51",
              "body1": "MuiTypography-body1-46",
              "body2": "MuiTypography-body2-45",
              "button": "MuiTypography-button-48",
              "caption": "MuiTypography-caption-47",
              "colorError": "MuiTypography-colorError-60",
              "colorInherit": "MuiTypography-colorInherit-56",
              "colorPrimary": "MuiTypography-colorPrimary-57",
              "colorSecondary": "MuiTypography-colorSecondary-58",
              "colorTextSecondary": "MuiTypography-colorTextSecondary-59",
              "display1": "MuiTypography-display1-41",
              "display2": "MuiTypography-display2-40",
              "display3": "MuiTypography-display3-39",
              "display4": "MuiTypography-display4-38",
              "gutterBottom": "MuiTypography-gutterBottom-54",
              "headline": "MuiTypography-headline-42",
              "noWrap": "MuiTypography-noWrap-53",
              "paragraph": "MuiTypography-paragraph-55",
              "root": "MuiTypography-root-37",
              "subheading": "MuiTypography-subheading-44",
              "title": "MuiTypography-title-43",
            }
          }
          color="default"
          gutterBottom={false}
          headlineMapping={
            Object {
              "body1": "p",
              "body2": "aside",
              "display1": "h1",
              "display2": "h1",
              "display3": "h1",
              "display4": "h1",
              "headline": "h1",
              "subheading": "h3",
              "title": "h2",
            }
          }
          noWrap={false}
          paragraph={false}
          variant="body1"
        >
          <p
            className="MuiTypography-root-37 MuiTypography-body1-46"
          >
            Unable to render attribute 
            Attribute 1
          </p>
        </Typography>
      </WithStyles(Typography)>
    </div>
  </AttributeDetails>
</WithStyles(AttributeDetails)>
`;
